docs/tutorial-smart-ai-1.md

```diff
@@ -0,0 +1,208 @@
+# Introduction to SmartAI
+
+## What type of waypoint will I use?
+
+[WIP]
+
+SmartAI allows us to use 2 different waypoint tables:
+
+waypoint_data, which is used in cpp and in more general patrols
+waypoints, a type which was made specifically for SmartAI
+
+[WIP]
+
+# Scripting a Simple Scripted Patrol
+
+For this we'll script a simple patrol RP, a script that plays every now and then on a timer and involves a path.
+
+When making blizzlike scripts, most of this data can be gathered in sniffs. Using tools like a parser, WDBE, or WaypointCreator we can extract waypoints, and then script in what actions are to be taken.
+
+In a parsed file, you can search for the unique creature GUID, and see what actions are done. For example, NPCFlag, UnitFlag, StandState, EmoteState and Oneshot Emotes all can be found in the parsed file. Some actions will eventually be hidden, but then we can get creative or even investigate further. In Black Temple, for example, there are creatures that execute an Eat Emote by casting a serverside spell, but that isn't common.
+
+The event we will be implementing contains only StandStates and spoken lines.
+
+First we detail what will happen and when:
+
+```
```

Comment:
consider adding the sql prefix to make the code within the codeblock get the correct colors